### PR TITLE
feat(server): add support for KEEPTTL and GET options with SET command

### PR DESCRIPTION
**Pull request by nmathew98**
Issue: #389 

Added support for KEEPTTL and GET options with SET command

### CODE REVIEW

**Comment by nmathew98** [comment]
@romange ready for review

File: src/server/string_family.cc [hunk 1]
```diff
@@ -550,8 +564,12 @@ void StringFamily::Set(CmdArgList args, ConnectionContext* cntx) {
       sparams.how = SetCmd::SET_IF_NOTEXIST;
     } else if (cur_arg == "XX") {
       sparams.how = SetCmd::SET_IF_EXISTS;
+    } else if (cur_arg == "GET") {
+      std::optional<std::string> prev_val = std::make_optional(std::string(""));
```
Comment (romange) [inline src/server/string_family.cc:568]: this is not good. you create a variable and then pass the pointer to it. The stack disappears.

File: src/server/string_family.cc [hunk 2]
```diff
@@ -444,6 +444,17 @@ OpStatus SetCmd::SetExisting(const SetParams& params, PrimeIterator it, ExpireIt
   PrimeValue& prime_value = it->second;
   EngineShard* shard = op_args_.shard;
 
+  DbSlice& db_slice = shard->db_slice();
+
+  uint64_t at_ms;
+  if (params.keep_expire) {
+    at_ms = std::max((long unsigned int) 0, db_slice.ExpireTime(e_it) - op_args_.db_cntx.time_now_ms);
```
Comment (romange) [inline src/server/string_family.cc:451]: it's not needed. if keep_expire is true why would you even call UpdateExpire below?

**Review by romange — COMMENTED** [review]
Hi @nmathew98 . Changing Dragonfly requires level of knowledge in C++ that you currently do not have. I suggest that you close this PR.

**Comment by nmathew98** [comment]
> Hi @nmathew98 . Changing Dragonfly requires level of knowledge in C++ that you currently do not have. I suggest that you close this PR.

Alright haha, thanks for your comments will try to improve it (no more pushes dw haha)!

**Pull request closed without merging**